tests/test_shell_completion.py

```diff
@@ -342,3 +344,25 @@ def test_choice_case_sensitive(value, expect):
     )
     completions = _get_words(cli, ["-a"], "a")
     assert completions == expect
+
+
+def test_add_completion_class():
+    _available_shells.clear()
```

Comment:
This is a risky move.

By clearing `_available_shells` you are clearing also all the pre-defined shells that's in there before the test.

A better way to go is to:
1. Save the previous content of `_available_shells` in a separted list
2. Clear `_available_shells`
3. Run the test
4. Insert back the shells you saved

In that way, your test will not effect other tests.
This is a perfect example for the use of [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html). If you are femiliar with those, you can create a fixture that uses the keyword `yield` for before-test and after-test effects.

Reply:
Thanks for your prompt review and for the suggestion. Indeed, clearing the dict of pre-defined shells gave me a bad feeling too... It was just a crude way to ensure these two tests didn't interfere with each other.

I didn't know about pytest fixtures before, but was glad to learn something! (And they seem very straightforward.) I've just pushed a new commit that uses a fixture, and hopefully that looks good to you as well.